test/unit/layers-metadata.spec.js

```diff
@@ -360,6 +360,157 @@ describe('layers metadata', () => {
+          typeIdField: null,
+          fields: [
+            {
+              name: 'OBJECTID',
+              type: 'esriFieldTypeOID',
+              alias: 'OBJECTID',
+              sqlType: 'sqlTypeInteger',
+              domain: null,
+              defaultValue: null,
+              editable: false,
+              nullable: false
+            }
+          ],
+          relationships: [],
+          capabilities: 'list,of,stuff',
+          maxRecordCount: 2000,
+          supportsStatistics: true,
+          supportsAdvancedQueries: true,
+          supportedQueryFormats: 'JSON',
+          ownershipBasedAccessControlForFeatures: {
+            allowOthersToQuery: true
+          },
+          useStandardizedQueries: true,
+          advancedQueryCapabilities: {
+            useStandardizedQueries: true,
+            supportsStatistics: true,
+            supportsOrderBy: true,
+            supportsDistinct: true,
+            supportsPagination: true,
+            supportsTrueCurve: false,
+            supportsReturningQueryExtent: true,
+            supportsQueryWithDistance: true
+          },
+          canModifyLayer: false,
+          dateFieldsTimeReference: null,
+          isDataVersioned: false,
+          supportsRollbackOnFailureParameter: true,
+          hasM: false,
+          hasZ: false,
+          allowGeometryUpdates: true,
+          objectIdField: 'OBJECTID',
+          globalIdField: '',
+          types: [],
+          templates: [],
+          hasStaticData: false,
+          timeInfo: {},
+          uniqueIdField: {
+            name: 'OBJECTID',
+            isSystemMaintained: true
+          },
+          currentVersion: 10.51,
+          fullVersion: '10.5.1',
+          minScale: 0,
+          maxScale: 0,
+          canScaleSymbols: false,
+          drawingInfo: {
+            renderer: {
+              type: 'simple',
+              symbol: {
+                color: [
+                  45,
+                  172,
+                  128,
+                  161
+                ],
+                outline: {
+                  color: [
+                    190,
+                    190,
+                    190,
+                    105
+                  ],
+                  width: 0.5,
+                  type: 'esriSLS',
+                  style: 'esriSLSSolid'
+                },
+                size: 7.5,
+                type: 'esriSMS',
+                style: 'esriSMSCircle'
+              }
+            },
+            labelingInfo: null
+          },
+          extent: {
+            xmin: -101,
+            xmax: -99,
+            ymin: 39,
+            ymax: 41,
+            spatialReference: {
+              wkid: 4326,
+              latestWkid: 4326
+            }
+          },
+          supportsCoordinatesQuantization: false,
+          hasLabels: false,
+          geometryType: 'esriGeometryPoint'
+        }
+      ],
+      tables: []
+    })
+  })
+
```

Comment:
Sorry, I think this test file is out of date and a lot of these tests are obsolete.  You don't need to bother testing `copyrightText` override here, because you're already testing table-layer-metadata.js directly.  Sorry for the trouble, but you can remove this test.  I'll note we need cleanup on the file.

Reply:
From a library consumer's perspective, this test is actually more important than the lower down tests. The only reason I wrote the feature-layer test is that I didn't have this test in place.

IMHO, tests like this (close to the API) are more important than the unit tests close to the implementation, because they inform library consumers how to use the library, and would make it easier for contributors to contribute. But that's just my opinion. If you don't mind my making a case for keeping these tests, I'd like to list the reasons why you might reconsider.

Here are my arguments:

1. Putting tests as close to the API as possible (tests being a safety net against future refactoring) not only clearly denote intention, but also give you (future refactorer) vast scope to refactor. The entire space behind the API is your playground!
2. Putting tests like this one close to the implementation (overriding a property, which don't really add to the value of the library, but are really a convenience for the consumer) makes it harder to move that implementation somewhere else: your task will then consist of not only the code refactoring, but finding the right place to put the tests. This is an important consideration if you have any chance of being wrong about your abstractions, which (let's face it) happens all the time.

All that said, I am VERY willing to comply with your request. I just prefer to at least try not to do something to another maintainer's library that I would never do in my own.

Thanks for your fast response! I really appreciate it!

Reply:
Yes, integration tests are fine.  Same comment as before though.  It's an integration test IMO not a unit test because it ends up testing the behavior of it'd dependencies.  Just leave it for now, and we'll move it to the integration test folder at some point.

Reply:
Ah! I guess I didn't dive into the project deep enough to see the separation of integration tests. I'll get the changes in shortly.

Reply:
Moved this one test out.